code/modules/sanity/breakdowns.dm

```diff
@@ -348,79 +348,253 @@
-	)
 	end_messages = list(
 		"You feel at ease again, suddenly."
 	)
 
 /datum/breakdown/common/obsession/can_occur()
 	var/list/candidates = (GLOB.player_list & GLOB.living_mob_list & GLOB.human_mob_list) - holder.owner
-	candidates = shuffle(candidates)
-	while(candidates.len)
-		var/mob/living/carbon/human/H = pick(candidates)
-		candidates.Remove(H)
-		var/list/contents = H.get_contents()
-		target = locate(/obj/item/weapon/oddity) in contents
-		if(!target)
-			target = pick(H.organs - H.organs_by_name[BP_CHEST])
-
-		if(target)
-			objectname = "[H.real_name]'s [target.name]"
-			break
-	return !!target
+	if(candidates.len)
+		target = pick(candidates)
+		start_messages = list("[target.name] knows the way out. [target.name] is hiding something. [target.name] is the key! [target.name] is yours!")
+		return TRUE
+	return FALSE
 
 /datum/breakdown/common/obsession/update()
 	. = ..()
 	if(!.)
-		return
-	if(QDELETED(target))
-		to_chat(holder.owner, SPAN_WARNING("\The [objectname] is lost!"))
-		finished = TRUE
-		conclude()
 		return FALSE
-	if(target.loc == holder.owner)
-		var/message = pick(list(
-			"Your mind convulses in the ecstasy. The sacred [objectname] is now yours!",
-			"You feel the warmth of the [objectname] in your head.",
-			"You suffered so long to achieve greatness! The sacred [objectname] is now yours. Only yours."
-		))
-		to_chat(holder.owner, SPAN_NOTICE(message))
-		holder.restoreLevel(70)
+	if(QDELETED(target))
+		to_chat(holder.owner, SPAN_WARNING("[target.name] is lost!"))
 		finished = TRUE
 		conclude()
 		return FALSE
-	if(world.time >= message_time)
-		message_time = world.time + OBSESSION_COOLDOWN
-		var/message = pick(list(
-			"You knew it. The [objectname] will ease your journey to the stars.",
-			"You look all around, but the only thing you can see is the [objectname].",
-			"Your thoughts are all about the [objectname].",
-			"You imagine how you will pour your hands into the still warm [objectname].",
-			"Vivid imagery of the [objectname] fills your brain.",
-			"You know it. It is the key to your salvation. [capitalize(objectname)]. [capitalize(objectname)]. [capitalize(objectname)]!",
-			"The voice within you demands only one thing: the [objectname].",
-			"It hurts you to keep pretending that your life without the [objectname] has meaning.",
-			"Your mind whispers to you with the only words in its silent throat: [objectname].",
-			"You know that only salvation from your sins is [objectname]."
-		))
-		to_chat(holder.owner, SPAN_NOTICE(message))
+	if(target in view(holder.owner))
+		delta_time += abs(world.time - last_time)
+		last_time = world.time
+		holder.owner.whisper_say("[target.name]")
+		if(delta_time >= obsession_time)
+			finished = TRUE
+			conclude()
+			return FALSE
+	else
+		last_time = world.time
+		if(world.time >= message_time)
+			message_time = world.time + BREAKDOWN_ALERT_COOLDOWN
+			var/message = pick(list("[target.name] knows the way out.",
```

Comment:
You should provide more messages. You something from rejected code above, for example. Also make it clear that user needs to see them and keep them in sight.